Prepare for AndroidX

NicolaVerbeeck opened the pull request.

NicolaVerbeeck committed:
Update to AndroidX

NicolaVerbeeck committed:
Workarounds for building on jitpack

blipinsk commented on library/build.gradle:
```diff
@@ -1,12 +1,12 @@
-    compileSdkVersion 23
-    buildToolsVersion "23.0.2"
+    compileSdkVersion 29
+    buildToolsVersion "29.0.2"
```
`buildToolsVersion` is no longer necessary

blipinsk commented on gradle.properties:
```diff
@@ -11,4 +11,6 @@ POM_LICENCE_NAME=The Apache Software License, Version 2.0
\ No newline at end of file
+POM_DEVELOPER_NAME=Bartosz Lipinski
+android.useAndroidX=true
+android.enableJetifier=true
```
why `android.enableJetifier=true`?

NicolaVerbeeck replied:
Very strange, in my local changes this is false and git does not see any changes...

blipinsk commented on sample/build.gradle:
```diff
@@ -1,13 +1,13 @@
-    compileSdkVersion 23
-    buildToolsVersion "23.0.2"
+    compileSdkVersion 29
+    buildToolsVersion "29.0.2"
```
same as before - no longer needed

blipinsk commented on library/src/main/java/com/bartoszlipinski/viewpropertyobjectanimator/PercentChangeListener.java:
```diff
@@ -17,7 +17,7 @@
 import android.animation.ValueAnimator.AnimatorUpdateListener;
-import android.support.percent.PercentLayoutHelper;
+import androidx.percentlayout.widget.PercentLayoutHelper;
 import android.view.View;
```
`Support` library is mentioned in the error messages in this class. Please modify them as well.

NicolaVerbeeck replied:
Do you mean the deprecation of PercentLayout?

blipinsk replied:
Those lines:
https://github.com/blipinsk/ViewPropertyObjectAnimator/blob/f0409606ce8fece46f6d90ec499478505e335699/library/src/main/java/com/bartoszlipinski/viewpropertyobjectanimator/PercentChangeListener.java#L46-L52

mention `part of the Percent Support Library`. 

This needs to be modified because it's no longer part of the **Support** library.

blipinsk commented on library/src/main/java/com/bartoszlipinski/viewpropertyobjectanimator/ViewPropertyObjectAnimator.java:
```diff
@@ -22,8 +22,8 @@
 import android.os.Build;
-import android.support.v4.util.ArrayMap;
-import android.support.v4.view.ViewCompat;
+import androidx.collection.ArrayMap;
```
Let's remove the whole `ArrayMap` (migrate to `HashMap`). 
The performance benefits with such small dataset is probably negligible and removing it will let us remove the `collection` dependency completely.

NicolaVerbeeck replied:
If we remove/replace the single method we take in from ViewCompat, we can drop support library in it's entirety (if we drop percent layout). What do you think?

blipinsk replied:
Removing `ViewCompat.isAttachedToWindow` is a good idea. Let's copy the internals of the method to the library.
As far as the percent library goes, let's keep it, especially since we use `compileOnly` with it.

blipinsk reviewed (changes requested):
Thanks for the Pull Request, here are a few notes.

Also, let's prepare the `CHANGELOG` for those changes (the new version will be `1.5.0`) & fix the `README` (`Support` mentioned).

blipinsk commented:
@NicolaChimerapps are you gonna fix those issues or should I take over?

NicolaVerbeeck commented:
I was going to take a look on Monday but feel free 😊

blipinsk commented:
> I was going to take a look on Monday but feel free 😊

Not sure if I can push to your fork, and I would like you to take credit for improving the library...

NicolaVerbeeck committed:
Undo specific changes to build.gradle
Remove direct dependency on any androidX libraries
Remove mention of support library and replace it with AndroidX
Replaced ArrayMap with HashMap
Replaced ViewCompat method with small compat copy

NicolaVerbeeck committed:
Update readme to reflect androidx

NicolaVerbeeck commented:
This should do it

Zhuinden commented:
What's up on this side?

blipinsk commented:
> What's up on this side?
 
Sorry for the delay @Zhuinden, the timing was unfortunate (I was extremely busy 😢). Let me try to finish this today 😉

blipinsk reviewed (approved):
I will be adding a few small details (like the Changelog), but overall thanks @NicolaChimerapps for your help here 😉
